testinfra/modules/pip.py

```diff
@@ -29,14 +29,16 @@ def get_packages(self, pip_path="pip"):
          'mywebsite': {'version': '1.0a3', 'path': '/srv/website'},
          'psycopg2': {'version': '2.6.2'}}
         """
-        output_re = re.compile(r'^(.+) \((.+)\)$')
+        output_re = re.compile(r'^(.+?)\s+(.+)$')
         pkgs = {}
-        out = self.check_output("%s list --no-index", pip_path)
-        for line in out.splitlines():
+        out = self.check_output(
+            "%s list --no-index --format=columns", pip_path
```

Comment:
Thanks for the patches, they work well.

But I just noticed that starting with pip 9 there is a `--format=json` option which will make things easier for us.
Also it's not a strong opinion but I'd like to keep compatibility with earlier pip version (those found on debian wheezy or centos 6). I think we can write something like this:

```
out = self.run_expect([0, 2], "pip list --no-index --format=columns")
if out.rc == 0:
    data = json.loads(out.stdout)
    # continue for pip >= 9
else:
    out = self.check_output("pip list --no-index")
    # continue for pip < 9
```

What do you think ?

Reply:
@philpep Thanks for the review.

It's funny you mention the `json` option, since I almost used that, then thought I'd keep it simple and stick with the text-based output. It could be a nice improvement for the future, but this seemed like a quicker route to regain pip 10.x compatibility.

Regarding the earlier pip versions, that seems good to me, but how do we test that? All the tests seemed to pass for my change, so does that mean we need to add some extra tests for the old distros you mentioned above?

Reply:
I think it's ok to not write tests for old pip version, you can just test manually for pip 8 and pip 7 and It will be fine to me.

Reply:
👍 I'll work on that and get back to you, then. Thanks.